client/human/HumanClientFSM.cpp

```diff
@@ -445,19 +469,23 @@ boost::statechart::result PlayingGame::react(const Error& msg) {
     bool fatal;
     ExtractMessageData(msg.m_message, problem, fatal);
 
-    ErrorLogger() << "PlayingGame::react(const Error& msg) error: " << problem;
+    ErrorLogger() << "PlayingGame::react(const Error& msg) error: "
+                  << problem << "\nProblem is" <<(fatal ? "":"non-")<<" fatal";
+
+    //Note: Any transit<> transition must occur before the MessageBox().
+    // MessageBox blocks and can allow other events to transit<> to a new state
+    // which makes this transit fatal.
+    boost::statechart::result retval = fatal ? transit<IntroMenu>() : discard_event();
     ClientUI::MessageBox(UserString(problem), true);
 
     if (fatal) {
-        Client().GetClientUI()->GetMessageWnd()->HandleGameStatusUpdate(UserString("RETURN_TO_INTRO") + "\n");
+        Client().EndGame(true);
```

Comment:
I renamed the AI script directory, then started freeorion, then quickstarted. I got a crash around here, though I'm now unable to reproduce it.